### ISSUE TITLE
Can't set cookie when using dynamic node names

### ISSUE THREAD

**Issue by isvilen** [erlang/otp#5402]
**Describe the bug**
When using dynamic node name call to `erlang:set_cookie` fails with `distribution_not_started` error.

**To Reproduce**
```
$ erl -sname undefined
Eshell V12.1.5  (abort with ^G)
1> erlang:set_cookie(cookie).
** exception error: distribution_not_started
     in function  auth:set_cookie/2 (auth.erl, line 124)
```

**Expected behavior**
To be able to set cookie after erlang node using dynamic node name is started .

**Affected versions**
Tested with 24.1.5, probably affects all 24+ versions 

**Additional context**
Call to `erlang:set_cookie(Node, Cookie)` using `node()` as first parameters also falis:
```
$ erl -sname undefined
Eshell V12.1.5  (abort with ^G)
1> erlang:set_cookie(node(), cookie).
** exception error: bad argument
     in function  erlang:set_cookie/2
        called as erlang:set_cookie(nonode@nohost,cookie)
        *** argument 1: the node name is not part of a distributed system
```

**Comment by sverker** [comment]
Thanks for the report. This is a known issue (OTP-16744).

**Comment by sverker** [comment]
Fix merged to maint for OTP 24.3.
